share/tasks.py

```diff
@@ -132,13 +132,15 @@ def run(self, normalized_id, started_by_id=None):
             started_by = ShareUser.objects.get(pk=started_by_id)
         logger.info('%s started make JSON patches for %s at %s', started_by, normalized, datetime.datetime.utcnow().isoformat())
 
+        cs = None
         try:
             cs = ChangeSet.objects.from_graph(ChangeGraph.from_jsonld(normalized.normalized_data, extra_namespace=normalized.source.username), normalized.id)
             if cs and (normalized.source.is_robot or normalized.source.is_trusted):
                 # TODO: verify change set is not overwriting user created object
                 cs.accept()
         except Exception as e:
-            logger.exception('Failed make json patches (%d)', normalized_id)
+            if cs and cs.id:
+                cs.delete()
```

Comment:
We decided against a transaction because I saw a deadlock that seemed like it might have been caused by wrapping all the disambiguation in a transaction, but now that I think about it that doesn't seem right... I'll put it in a transaction to be sure all the changes are rolled back, but we should keep an eye on it under load.